public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx

```diff
@@ -146,6 +146,18 @@ export const ConnectionSVG = ({
             yStart = v.sourceOriginal.y;
             xEnd = v.targetOriginal.x;
             yEnd = v.targetOriginal.y;
+          } else if (source.options.connections) {
+            const currentConnections = [...source.options.connections];
+            // Check for original state
+            if (!currentConnections[index].sourceOriginal || !currentConnections[index].targetOriginal) {
+              currentConnections[index] = {
+                ...currentConnections[index],
+                sourceOriginal: { x: x1, y: y1 },
+                targetOriginal: { x: x2, y: y2 },
+              };
+              // Update model with populated originals
+              source.onChange({ ...source.options, connections: currentConnections });
```

Comment:
This onChange call is causing the warning.

Reply:
I don't suppose it's as easy as building a helper function that calls the onChange, then calling he helper function inside the render? Little bob an' weave? 😂

Reply:
Good thought, but that didn't work.

Reply:
It looks like we can avoid using this `onChange` call all together and just modify the state directly and avoid this issue, with this I think we can move forward and merge this PR as solution is "good enough" 😅